keps/sig-storage/2639-secret-protection/README.md

```diff
@@ -0,0 +1,428 @@
+
+### Non-Goals
+
+- Protect important secrets that aren't in use from deleting
+- Protect other resources than secret from deleting.
+
+## Proposal
+
+A new controller to protect secret is introduced.
+
+### User Stories
+
+#### Story 1
+
+A user creates a secret and a pod using the secret. Then, the user mistakenly delete the secret while the pod is using it.
+The secret is protected until the pod using the secret is deleted.
+
+#### Story 2
+
+A user creates a volume that uses a certain secret in the same namespace. Then, the user delete the namespace.
+The secret is protected until the volume using the secret is deleted and the deletion of the volume succeeds.
+
+#### Story 3
+
+A user really would like to delete a secret inspite that it is used by other resources.
+The user force delete the secret while it is used by other resources, and the secret isn't protected and is actually deleted.
+
+### Notes/Constraints/Caveats (Optional)
+
+- Compatibility:
+  - There might be many existing scripts that don't care the order of deletion. Therefore, such scripts might stuck on secret deletion, if the deletion of the resources using the secrets are done later.
+- Usability:
+  - Use of the secret in other resource will not be obvious to users. Therefore, users might not easily understand why the secret is not deleted.
+  - Users might need to force delete the secret on deletion or would like to avoid protection for certain secrets that already exist or that are newly created.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+This feature would be able to implement by the same way to pv-protection/pvc-protection.
+- The deletion is blocked by using newly introduced `kubernetes.io/secret-protection` finalizer,
+- The finalizer will be added on creation of the secret by using admission controller,
+- The finalizer will be deleted by newly introduced `secret-protection-controller` by checking whether the secret is in-use, on every change(Create/Update/Delete) events for secrets and related resrouces.
+
+### Test Plan
+
+- For Alpha, unit tests and e2e tests verifying that a secret used by other resources is protected by this feature are added.
+- For Beta, scalability tests are added to exercise this feature.
+- For GA, the introduced e2e tests will be promoted to conformance.
+
+### Graduation Criteria
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- Allowing time for feedback
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
```

Comment:
Which flag are you referring to?

Reply:
Actually, I just moved this from the template. I guess that this is feature flag. If this is command line flag or something like that, we won't need this entire section.

Reply:
I'd suggest just say "N/A" for this section or simply remove it.